File: memento/exercise/History.java
package memento.exercise;

import memento.TextEditorState;

import java.util.Stack;

public class History {

    private final Stack<CanvasState> prevStates;

    public History() {
        this.prevStates = new Stack<>();
    }

    public void saveHistoryState(CanvasState canvasState) {
        prevStates.push(canvasState);
    }

    public CanvasState undo() {
        if (!prevStates.isEmpty()) {
            return prevStates.pop();
        }
        return null;
    }
}

File: C#/Memento/Exercise/History.cs
namespace Memento.Exercise;

public sealed class History
{
    private readonly Stack<CanvasState> _prevStates;

    public History()
        => _prevStates = new Stack<CanvasState>();

    public void SaveHistoryState(CanvasState canvasState)
        => _prevStates.Push(canvasState);

    public CanvasState? Undo()
        => _prevStates.Count > 0 ? _prevStates.Pop() : null;
}
